app/views/layouts/exercise_panel/_exercise.html.erb

```diff
@@ -6,6 +6,11 @@
     <%= exercise.submissions_count %>
   </td>
   <td>
+    <% if exercise.solved_by?(current_user) && current_user %>
+      <span class="glyphicon glyphicon-certificate passed-exercise" aria-hidden="true"></span>
+    <% elsif !exercise.solved_by?(current_user) && current_user %>
```

Comment:
Aside of previous comments, I like this. Could we extract to a helper? That way we would not need to repeat that logic, and we can test it.